src/socket/tcp.rs

```diff
@@ -2516,6 +2580,53 @@ impl<'a> Socket<'a> {
                 .unwrap_or(&PollAt::Ingress)
         }
     }
+
+    /// Replace the receive buffer with a new one.
+    ///
+    /// The requirements for the new buffer are:
+    /// 1. The new buffer must be larger than the length of remaining data in the current buffer
+    /// 2. The new buffer must be multiple of (1 << self.remote_win_shift)
+    ///
+    /// If the new buffer does not meet the requirements, the new buffer is returned as an error;
+    /// otherwise, the old buffer is returned as an Ok value.
+    ///
+    /// See also the [local_recv_win_scale](struct.Socket.html#method.local_recv_win_scale) methods.
+    pub fn replace_recv_buffer<T: Into<SocketBuffer<'a>>>(
```

Comment:
This looks like a generic method that has no direct correspondence to TCP receive window resizing and as such I don't see why it should be a part of this PR at all.

Reply:
This method does enable the TCP receive window resizing. Smoltcp uses `rx_buffer.capacity()-rx_buffer.len()` to compute the current receive window. Without this method, it's impossible to adjust `rx_buffer.capacity()`. By replaced with a larger receive buffer, smoltcp knows it can advertise a larger receive window size safely without correctness issue.